packages/flutter/lib/src/material/reorderable_list.dart

```diff
@@ -584,7 +597,29 @@ class _ReorderableListContentState extends State<_ReorderableListContent> with T
             if (widget.reverse) _wrap(finalDropArea, widget.children.length, constraints),
             if (widget.header != null) widget.header,
             for (int i = 0; i < widget.children.length; i += 1) _wrap(widget.children[i], i, constraints),
-            if (!widget.reverse) _wrap(finalDropArea, widget.children.length, constraints),
+            if (widget.footer != null)
+              // We need to wrap the widget.footer so we can reorder an item between
+              // the last item of the list and the footer itself. If the footer
+              // is used instead of the finalDropArea it needs its own key,
+              // because _wrap() requires it.
+              _wrap(ConstrainedBox(
```

Comment:
I don't quite understand what is going on here. How come the header requires the `ConstrainedBox` w/ `BoxCosntraints`, as well as the `Align` widget, whereas the header has none of these things?

I would expect that the header and footer should be rather symmetric in terms of implementation since they're basically the same thing, but on opposite ends of the ReorderableListView